torch/_inductor/fx_passes/group_batch_fusion.py

```diff
@@ -156,6 +156,116 @@ def fuse(self, graph, subset):
             graph.erase_node(original_mm)
 
 
+class BatchLayernormFusion(BatchFusion):
+    """
+    Batch layer norm fusion in pre grad pass
+    """
+    def is_weight_valid(self, weight):
+        if weight is None:
+            return True
+        else:
+            if "example_value" in weight.meta:
+                return True
+            else:
+                return False
+
+    def match(self, node):
+        if CallFunctionVarArgs(torch.nn.functional.layer_norm).match(node):
+            input = get_arg_value(node, 0, "input")
+            weight = get_arg_value(node, 2, "weight")
+            bias = get_arg_value(node, 3, "bias")
+            group_key = (
+                (
+                    "batch_layernorm",
+                    str(input.meta["example_value"].shape),
+                    str(weight.meta["example_value"].shape) if weight is not None else "",
+                    str(get_arg_value(node, 1, "normalized_shape")),
+                    weight is None,
+                    bias is None,
```

Comment:
Do these need to be shapes too?  

Not sure if things like broadcasting are legal.